src/map/lua/luautils.cpp

```diff
@@ -5825,4 +5826,65 @@ namespace luautils
 
         return table;
     }
+
+    // Determines if a table is valid to use ipairs()
+    // Also implies #tableName reliably returns the size of the table
+    bool luautils::IsTableIpairsCompatible(const sol::optional<sol::table>& maybeSequentialTable)
+    {
+        if (!maybeSequentialTable)
+        {
+            return false;
+        }
+
+        sol::table t = *maybeSequentialTable;
```

Comment:
Like in the other PR: I don't know what the semantics of `sol::table` are, this may very well be a copy. Experiment with this being a reference - just in case (unless you want to dig into the guts of `sol::table` to find out the truth)